main/lsp/ErrorReporter.h

```diff
@@ -0,0 +1,34 @@
+#ifndef RUBY_TYPER_LSP_ERRORREPORTER_H
+#define RUBY_TYPER_LSP_ERRORREPORTER_H
+
+#include "core/core.h"
+
+namespace sorbet::realmain::lsp {
+class LSPConfiguration;
+struct ErrorStatus {
+    // The epoch at which we last sent diagnostics for this file.
+    u4 lastReportedEpoch = 0;
+    // If true, the client believes this file has errors.
+    bool hasErrors = false;
+};
+class ErrorReporter {
+    const std::shared_ptr<const LSPConfiguration> config;
+    // Maps from file ref ID to its error status.
+    std::vector<ErrorStatus> fileErrorStatuses;
+    void setMaxFileId(u4 id);
```

Comment:
Don't forget to remove this declaration!

Reply:
🤦‍♂️